File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/TagJpaRepository.java
package com.cakk.domain.mysql.repository.jpa;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.cakk.domain.mysql.entity.cake.Tag;

@Repository
public interface TagJpaRepository extends JpaRepository<Tag, Long> {

	Optional<Tag> findTagByTagName(String tagName);
}

File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/TagJpaRepository.kt
package com.cakk.domain.mysql.repository.jpa

import com.cakk.domain.mysql.entity.cake.Tag
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface TagJpaRepository : JpaRepository<Tag, Long> {

	fun findTagByTagName(tagName: String?): Tag?
    fun findTagsByTagNameIsIn(tagNames: List<String>): List<Tag>
}

